tests: uart: sample.boards.stm32.uart.circular_dma : build failure for multiply NXP platforms

**Describe the bug**

sample.boards.stm32.uart.circular_dma build failure for below platforms in NXP

imx95_evk/mimx9596/m7
frdm_mcxc444/mcxc444
frdm_mcxw71/mcxw716c
frdm_mcxc242/mcxc242
frdm_ke15z/mke15z7
frdm_kw41z/mkw41z4

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - NXP platform
 - What have you tried to diagnose or workaround this issue?
 - none
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
this issue is introduced by
```
 e0da2d917fd73599fa3c67dcd71f0217891d46bd

samples: boards: st: uart: add sample for uart circular mode

This code example  will demonstrate how to use uart
circular mode on STM32 boards.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>
```

**To Reproduce**


```
scripts/twister -p imx95_evk/mimx9596/m7 -T samples/boards/st/uart
```




**Expected behavior**

this samples shall be filter out for none supported platform

**Impact**
CI build

**Logs and console output**

```
Loading Zephyr module(s) (Zephyr base): zephyr_default:dts -- Application: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/st/uart/circular_dma -- CMake version: 3.22.1 -- Found Python3: /usr/bin/python3 (found suitable version "3.10.12", minimum required is "3.10") found components: Interpreter -- Cache files will be written to: /home/user/.cache/zephyr -- Zephyr version: 4.0.99 (/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr) -- Found west (found suitable version "1.3.0", minimum required is "0.14.0") -- Board: frdm_kw41z, qualifiers: mkw41z4 -- Found host-tools: zephyr 0.17.0 (/opt/toolchains/zephyr-sdk-0.17.0) -- Found toolchain: zephyr 0.17.0 (/opt/toolchains/zephyr-sdk-0.17.0) -- Found Dtc: /opt/toolchains/zephyr-sdk-0.17.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") -- Found BOARD.dts: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/boards/nxp/frdm_kw41z/frdm_kw41z.dts -- Generated zephyr.dts: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/zephyr.dts -- Generated pickled edt: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/edt.pickle -- Generated zephyr.dts: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/zephyr.dts -- Generated devicetree_generated.h: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h -- Including generated dts.cmake file: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/dts.cmake Loading Zephyr default modules (Zephyr base). -- Application: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/st/uart/circular_dma -- CMake version: 3.22.1 -- Found Python3: /usr/bin/python3 (found suitable version "3.10.12", minimum required is "3.10") found components: Interpreter -- Cache files will be written to: /home/user/.cache/zephyr -- Zephyr version: 4.0.99 (/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr) -- Found west (found suitable version "1.3.0", minimum required is "0.14.0") -- Board: frdm_kw41z, qualifiers: mkw41z4 -- Found host-tools: zephyr 0.17.0 (/opt/toolchains/zephyr-sdk-0.17.0) -- Found toolchain: zephyr 0.17.0 (/opt/toolchains/zephyr-sdk-0.17.0) -- Found Dtc: /opt/toolchains/zephyr-sdk-0.17.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") -- Found BOARD.dts: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/boards/nxp/frdm_kw41z/frdm_kw41z.dts -- Generated zephyr.dts: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/zephyr.dts -- Generated pickled edt: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/edt.pickle -- Generated zephyr.dts: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/zephyr.dts -- Generated devicetree_generated.h: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h -- Including generated dts.cmake file: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/dts.cmake Parsing /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/Kconfig Loaded configuration '/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/boards/nxp/frdm_kw41z/frdm_kw41z_defconfig' Merged configuration '/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/st/uart/circular_dma/prj.conf' Merged configuration '/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/misc/generated/extra_kconfig_options.conf' Configuration saved to '/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/.config' Kconfig header saved to '/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/autoconf.h' -- Found GnuLd: /opt/toolchains/zephyr-sdk-0.17.0/arm-zephyr-eabi/arm-zephyr-eabi/bin/ld.bfd (found version "2.38") -- The C compiler identification is GNU 12.2.0 -- The CXX compiler identification is GNU 12.2.0 -- The ASM compiler identification is GNU -- Found assembler: /opt/toolchains/zephyr-sdk-0.17.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc Load components for MKW41Z4: driver_common component is included. driver_reset component is included. device_CMSIS component is included. CMSIS_Include_core_cm component is included. device_system component is included. driver_lpuart component is included. driver_port component is included. -- Using ccache: /usr/local/bin/ccache CMake Warning at /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/CMakeLists.txt:1002 (message): No SOURCES given to Zephyr library: drivers__dma Excluding target from build. -- Configuring done -- Generating done CMake Warning: Manually-specified variables were not used by the project: TC_NAME TC_RUNID -- Build files have been written to: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma [1/129] Generating include/generated/zephyr/version.h -- Zephyr version: 4.0.99 (/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr), build: v4.0.0-3190-g99a63a776980 [2/129] Preparing syscall dependency handling [3/129] Generating misc/generated/syscalls_subdirs.trigger [4/129] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json [5/129] Generating include/generated/zephyr/syscall_dispatch.c, include/generated/zephyr/syscall_exports_llext.c, syscall_weakdefs_llext.c, include/generated/zephyr/syscall_list.h [6/129] Generating include/generated/zephyr/driver-validation.h [7/129] Generating include/generated/zephyr/kobj-types-enum.h, include/generated/zephyr/otype-to-str.h, include/generated/zephyr/otype-to-size.h [8/129] Generating include/generated/device-api-sections.ld, include/generated/device-api-sections.cmake [9/129] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj [10/129] Generating include/generated/zephyr/offsets.h [11/129] Building C object CMakeFiles/app.dir/src/main.c.obj [12/129] Linking C static library app/libapp.a [13/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/heap/heap.c.obj [14/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj [15/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj [16/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj [17/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj [18/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj [19/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj [20/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/dec.c.obj [21/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/hex.c.obj [22/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/rb.c.obj [23/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/timeutil.c.obj [24/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/bitarray.c.obj [25/129] Building C object zephyr/CMakeFiles/zephyr.dir/lib/utils/ring_buffer.c.obj [26/129] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj [27/129] Building C object zephyr/CMakeFiles/zephyr.dir/soc/nxp/kinetis/flash_configuration.c.obj [28/129] Building C object zephyr/CMakeFiles/zephyr.dir/soc/nxp/kinetis/kwx/soc_kw4xz.c.obj [29/129] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj [30/129] Linking C static library zephyr/libzephyr.a [31/129] Generating linker_zephyr_pre0.cmd [32/129] Building C object zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj [33/129] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj [34/129] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj [35/129] Linking C static library zephyr/arch/common/libisr_tables.a [36/129] Linking C static library zephyr/arch/common/libarch__common.a [37/129] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/fatal.c.obj [38/129] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/nmi.c.obj [39/129] Building ASM object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/nmi_on_reset.S.obj [40/129] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/tls.c.obj [41/129] Linking C static library zephyr/arch/arch/arm/core/libarch__arm__core.a [42/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/exc_exit.c.obj [43/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fault.c.obj [44/129] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fault_s.S.obj [45/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fpu.c.obj [46/129] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/reset.S.obj [47/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/scb.c.obj [48/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/thread_abort.c.obj [49/129] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/vector_table.S.obj [50/129] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/swap_helper.S.obj [51/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_manage.c.obj [52/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/prep_c.c.obj [53/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/thread.c.obj [54/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cpu_idle.c.obj [55/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_init.c.obj [56/129] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/isr_wrapper.c.obj [57/129] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/__aeabi_read_tp.S.obj [58/129] Linking C static library zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a [59/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/assert.c.obj [60/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/cbprintf.c.obj [61/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/chk_fail.c.obj [62/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/errno_wrap.c.obj [63/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/exit.c.obj [64/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/locks.c.obj [65/129] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/stdio.c.obj [66/129] Linking C static library zephyr/lib/libc/picolibc/liblib__libc__picolibc.a [67/129] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/abort.c.obj [68/129] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/malloc.c.obj [69/129] Linking C static library zephyr/lib/libc/common/liblib__libc__common.a [70/129] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_mcg.c.obj [71/129] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_sim.c.obj [72/129] Linking C static library zephyr/drivers/clock_control/libdrivers__clock_control.a [73/129] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj [74/129] Linking C static library zephyr/drivers/console/libdrivers__console.a [75/129] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_mcux.c.obj [76/129] Linking C static library zephyr/drivers/gpio/libdrivers__gpio.a [77/129] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj [78/129] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_nxp_port.c.obj [79/129] Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a [80/129] Building C object zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj FAILED: zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj ccache /opt/toolchains/zephyr-sdk-0.17.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DCPU_MKW41Z512VHT4 -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DNDEBUG -DPICOLIBC_LONG_LONG_PRINTF_SCANF -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/soc/nxp/kinetis -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/soc/nxp/kinetis/kwx/. -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/soc/nxp/kinetis/common -I/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/cmsis/CMSIS/Core/Include -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/modules/cmsis/. -I/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/devices/MKW41Z4 -I/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/devices/MKW41Z4/drivers -I/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/drivers/lpuart -I/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk/drivers/port -I/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/modules/hal_nxp/. -isystem /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/lib/libc/common/include -Wshadow -fno-strict-aliasing -Os -imacros /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0plus -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/opt/toolchains/zephyr-sdk-0.17.0/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/samples/boards/st/uart/circular_dma=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=picolibc.specs -std=c99 -MD -MT zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj -MF zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj.d -o zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_mcux_lpuart.c.obj -c /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c: In function 'mcux_lpuart_configure_async': /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1070:16: warning: implicit declaration of function 'LPUART_GetRxFifoCount' [-Wimplicit-function-declaration] 1070 | while (LPUART_GetRxFifoCount(config->base) > 0) { | ^~~~~~~~~~~~~~~~~~~~~ In file included from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:34, from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/irq_multilevel.h:16, from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:20, from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:12, from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:11: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c: At top level: /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:96:41: error: '__device_dts_ord_DT_N_S_soc_S_lpuart_40054000_P_dmas_NAME_rx_PH_ORD' undeclared here (not in a function) 96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id) | ^~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4946:9: note: in expansion of macro 'COND_CODE_1' 4946 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/toolchain/common.h:138:23: note: in expansion of macro '_DO_CONCAT' 138 | #define _CONCAT(x, y) _DO_CONCAT(x, y) | ^~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:96:33: note: in expansion of macro '_CONCAT' 96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:237:37: note: in expansion of macro 'DEVICE_NAME_GET' 237 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id)) | ^~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:254:34: note: in expansion of macro 'DEVICE_DT_NAME_GET' 254 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id)) | ^~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1324:25: note: in expansion of macro 'DEVICE_DT_GET' 1324 | DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(id, rx)), \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1379:9: note: in expansion of macro 'RX_DMA_CONFIG' 1379 | RX_DMA_CONFIG(n) \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1390:9: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG' 1390 | LPUART_MCUX_DECLARE_CFG(n) \ | ^~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:7561:45: note: in expansion of macro 'LPUART_MCUX_INIT' 7561 | #define DT_FOREACH_OKAY_INST_nxp_lpuart(fn) fn(0) | ^~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1401:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 1401 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:5266:32: error: 'DT_N_S_soc_S_lpuart_40054000_P_dmas_NAME_rx_VAL_mux' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_lpuart_40054000_IRQ_IDX_0_VAL_irq'? 5266 | #define DT_N_INST_0_nxp_lpuart DT_N_S_soc_S_lpuart_40054000 | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4946:9: note: in expansion of macro 'COND_CODE_1' 4946 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1650:9: note: in expansion of macro 'DT_CAT7' 1650 | DT_CAT7(node_id, _P_, pha, _NAME_, name, _VAL_, cell) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:221:9: note: in expansion of macro 'DT_PHA_BY_NAME' 221 | DT_PHA_BY_NAME(node_id, dmas, name, cell) | ^~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:9: note: in expansion of macro 'DT_DMAS_CELL_BY_NAME' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT' 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__) | ^~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:339:31: note: in expansion of macro 'UTIL_CAT' 339 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat)) | ^~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3869:27: note: in expansion of macro 'DT_INST' 3869 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:30: note: in expansion of macro 'DT_DRV_INST' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1326:25: note: in expansion of macro 'DT_INST_DMAS_CELL_BY_NAME' 1326 | DT_INST_DMAS_CELL_BY_NAME(id, rx, mux), \ | ^~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1379:9: note: in expansion of macro 'RX_DMA_CONFIG' 1379 | RX_DMA_CONFIG(n) \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1390:9: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG' 1390 | LPUART_MCUX_DECLARE_CFG(n) \ | ^~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:7561:45: note: in expansion of macro 'LPUART_MCUX_INIT' 7561 | #define DT_FOREACH_OKAY_INST_nxp_lpuart(fn) fn(0) | ^~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1401:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 1401 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:5266:32: error: 'DT_N_S_soc_S_lpuart_40054000_P_dmas_NAME_rx_VAL_source' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_lpuart_40054000_P_wakeup_source'? 5266 | #define DT_N_INST_0_nxp_lpuart DT_N_S_soc_S_lpuart_40054000 | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4946:9: note: in expansion of macro 'COND_CODE_1' 4946 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1650:9: note: in expansion of macro 'DT_CAT7' 1650 | DT_CAT7(node_id, _P_, pha, _NAME_, name, _VAL_, cell) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:221:9: note: in expansion of macro 'DT_PHA_BY_NAME' 221 | DT_PHA_BY_NAME(node_id, dmas, name, cell) | ^~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:9: note: in expansion of macro 'DT_DMAS_CELL_BY_NAME' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT' 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__) | ^~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:339:31: note: in expansion of macro 'UTIL_CAT' 339 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat)) | ^~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3869:27: note: in expansion of macro 'DT_INST' 3869 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:30: note: in expansion of macro 'DT_DRV_INST' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1338:37: note: in expansion of macro 'DT_INST_DMAS_CELL_BY_NAME' 1338 | .dma_slot = DT_INST_DMAS_CELL_BY_NAME( \ | ^~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1379:9: note: in expansion of macro 'RX_DMA_CONFIG' 1379 | RX_DMA_CONFIG(n) \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1390:9: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG' 1390 | LPUART_MCUX_DECLARE_CFG(n) \ | ^~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:7561:45: note: in expansion of macro 'LPUART_MCUX_INIT' 7561 | #define DT_FOREACH_OKAY_INST_nxp_lpuart(fn) fn(0) | ^~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1401:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 1401 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:96:41: error: '__device_dts_ord_DT_N_S_soc_S_lpuart_40054000_P_dmas_NAME_tx_PH_ORD' undeclared here (not in a function) 96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id) | ^~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4946:9: note: in expansion of macro 'COND_CODE_1' 4946 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/toolchain/common.h:138:23: note: in expansion of macro '_DO_CONCAT' 138 | #define _CONCAT(x, y) _DO_CONCAT(x, y) | ^~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:96:33: note: in expansion of macro '_CONCAT' 96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:237:37: note: in expansion of macro 'DEVICE_NAME_GET' 237 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id)) | ^~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/device.h:254:34: note: in expansion of macro 'DEVICE_DT_NAME_GET' 254 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id)) | ^~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1301:25: note: in expansion of macro 'DEVICE_DT_GET' 1301 | DEVICE_DT_GET(DT_INST_DMAS_CTLR_BY_NAME(id, tx)), \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1380:9: note: in expansion of macro 'TX_DMA_CONFIG' 1380 | TX_DMA_CONFIG(n) \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1390:9: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG' 1390 | LPUART_MCUX_DECLARE_CFG(n) \ | ^~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:7561:45: note: in expansion of macro 'LPUART_MCUX_INIT' 7561 | #define DT_FOREACH_OKAY_INST_nxp_lpuart(fn) fn(0) | ^~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1401:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 1401 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:5266:32: error: 'DT_N_S_soc_S_lpuart_40054000_P_dmas_NAME_tx_VAL_mux' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_lpuart_40054000_IRQ_IDX_0_VAL_irq'? 5266 | #define DT_N_INST_0_nxp_lpuart DT_N_S_soc_S_lpuart_40054000 | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4946:9: note: in expansion of macro 'COND_CODE_1' 4946 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1650:9: note: in expansion of macro 'DT_CAT7' 1650 | DT_CAT7(node_id, _P_, pha, _NAME_, name, _VAL_, cell) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:221:9: note: in expansion of macro 'DT_PHA_BY_NAME' 221 | DT_PHA_BY_NAME(node_id, dmas, name, cell) | ^~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:9: note: in expansion of macro 'DT_DMAS_CELL_BY_NAME' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT' 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__) | ^~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:339:31: note: in expansion of macro 'UTIL_CAT' 339 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat)) | ^~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3869:27: note: in expansion of macro 'DT_INST' 3869 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:30: note: in expansion of macro 'DT_DRV_INST' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1303:25: note: in expansion of macro 'DT_INST_DMAS_CELL_BY_NAME' 1303 | DT_INST_DMAS_CELL_BY_NAME(id, tx, mux), \ | ^~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1380:9: note: in expansion of macro 'TX_DMA_CONFIG' 1380 | TX_DMA_CONFIG(n) \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1390:9: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG' 1390 | LPUART_MCUX_DECLARE_CFG(n) \ | ^~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:7561:45: note: in expansion of macro 'LPUART_MCUX_INIT' 7561 | #define DT_FOREACH_OKAY_INST_nxp_lpuart(fn) fn(0) | ^~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1401:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 1401 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:5266:32: error: 'DT_N_S_soc_S_lpuart_40054000_P_dmas_NAME_tx_VAL_source' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_lpuart_40054000_P_wakeup_source'? 5266 | #define DT_N_INST_0_nxp_lpuart DT_N_S_soc_S_lpuart_40054000 | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET' 72 | #define __DEBRACKET(...) __VA_ARGS__ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET' 64 | __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE' 59 | __COND_CODE(_XXXX##_flag, _if_1_code, _else_code) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_macro.h:204:9: note: in expansion of macro 'Z_COND_CODE_1' 204 | Z_COND_CODE_1(_flag, _if_1_code, _else_code) | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:4946:9: note: in expansion of macro 'COND_CODE_1' 4946 | COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \ | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:1650:9: note: in expansion of macro 'DT_CAT7' 1650 | DT_CAT7(node_id, _P_, pha, _NAME_, name, _VAL_, cell) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:221:9: note: in expansion of macro 'DT_PHA_BY_NAME' 221 | DT_PHA_BY_NAME(node_id, dmas, name, cell) | ^~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:9: note: in expansion of macro 'DT_DMAS_CELL_BY_NAME' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_N_INST_0_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:104:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT' 104 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__) | ^~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:339:31: note: in expansion of macro 'UTIL_CAT' 339 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat)) | ^~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree.h:3869:27: note: in expansion of macro 'DT_INST' 3869 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT) | ^~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/devicetree/dma.h:252:30: note: in expansion of macro 'DT_DRV_INST' 252 | DT_DMAS_CELL_BY_NAME(DT_DRV_INST(inst), name, cell) | ^~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1315:37: note: in expansion of macro 'DT_INST_DMAS_CELL_BY_NAME' 1315 | .dma_slot = DT_INST_DMAS_CELL_BY_NAME( \ | ^~~~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1380:9: note: in expansion of macro 'TX_DMA_CONFIG' 1380 | TX_DMA_CONFIG(n) \ | ^~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1390:9: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG' 1390 | LPUART_MCUX_DECLARE_CFG(n) \ | ^~~~~~~~~~~~~~~~~~~~~~~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/twister-out/frdm_kw41z_mkw41z4/zephyr/samples/boards/st/uart/circular_dma/sample.boards.stm32.uart.circular_dma/zephyr/include/generated/zephyr/devicetree_generated.h:7561:45: note: in expansion of macro 'LPUART_MCUX_INIT' 7561 | #define DT_FOREACH_OKAY_INST_nxp_lpuart(fn) fn(0) | ^~ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpuart' 105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__ | ^ /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/serial/uart_mcux_lpuart.c:1401:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY' 1401 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT) | ^~~~~~~~~~~~~~~~~~~~~~~~~~~ ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v4.0.0-3190-g99a63a776980

--- a/samples/boards/st/uart/circular_dma/sample.yaml
+++ b/samples/boards/st/uart/circular_dma/sample.yaml
@@ -7,5 +7,5 @@ tests:
     tags:
       - serial
       - uart
-    filter: dt_chosen_enabled("zephyr,shell-uart")
+    filter: dt_chosen_enabled("zephyr,shell-uart") and CONFIG_SOC_FAMILY_STM32
     harness: keyboard

